src/WarcraftLegacies.Source/Quests/KulTiras/QuestUnlockShip.cs

```diff
@@ -1,66 +1,98 @@
 
-    protected override string CompletionPopup => "The Proudmoore capital ship is now ready to sails!";
+    /// <inheritdoc/>
+    protected override string CompletionPopup => "The Proudmoore Capital Ship is now ready to set sail!";
 
+    /// <inheritdoc/>
     protected override string RewardDescription =>
       "Unpause the Proudmoore capital ship and unlocks the buildings inside.";
 
+    /// <inheritdoc/>
     protected override void OnComplete(Faction completingFaction)
     {
-      foreach (var unit in _rescueUnits) unit.Rescue(completingFaction.Player);
-      PauseUnit(_proudmooreCapitalShip, false);
+      if (completingFaction.Player != null)
+      {
+        completingFaction.Player.RescueGroup(_rescueUnits);
+        _proudmooreCapitalShip.Rescue(completingFaction.Player);
+      }
+      else
+      {
+        Player(bj_PLAYER_NEUTRAL_VICTIM).RescueGroup(_rescueUnits);
+        _proudmooreCapitalShip.Rescue(Player(PLAYER_NEUTRAL_AGGRESSIVE));
+      }
+      PauseUnitBJ(false, _proudmooreCapitalShip);
+      KultirasSetup.Kultiras?.Player?.SetTeam(TeamSetup.Alliance);
+      ZandalarSetup.Zandalar?.Player?.SetTeam(TeamSetup.Horde);
     }
 
+    /// <inheritdoc/>
     protected override void OnFail(Faction completingFaction)
     {
       LegendKultiras.LegendKatherine.StartingXp = GetHeroXP(LegendKultiras.LegendKatherine.Unit);
       completingFaction.Obliterate();
-      LegendKultiras.LegendKatherine.ForceCreate(completingFaction.Player, new Point(-15223, -22856), 110);
-      UnitAddItem(LegendKultiras.LegendKatherine.Unit,
-        CreateItem(FourCC("I00M"), GetUnitX(LegendKultiras.LegendKatherine.Unit),
-          GetUnitY(LegendKultiras.LegendKatherine.Unit)));
-      if (GetLocalPlayer() == completingFaction.Player)
-        SetCameraPosition(Regions.ShipAmbient.Center.X, Regions.ShipAmbient.Center.Y);
-      foreach (var unit in _rescueUnits) unit.Rescue(completingFaction.Player);
-      PauseUnit(_proudmooreCapitalShip, true);
-      SetUnitOwner(_proudmooreCapitalShip, completingFaction.Player, true);
+      if (completingFaction.Player != null)
+      {
+        LegendKultiras.LegendKatherine.ForceCreate(completingFaction.Player, new Point(_rescueRect.Center.X, _rescueRect.Center.Y), 110);
+        LegendKultiras.LegendAdmiral.ForceCreate(completingFaction.Player, new Point(_rescueRect.Center.X, _rescueRect.Center.Y), 110);
+        if (GetLocalPlayer() == completingFaction.Player)
+          SetCameraPosition(_rescueRect.Center.X, _rescueRect.Center.Y);
+        completingFaction.Player.RescueGroup(_rescueUnits);
+        completingFaction.Player.AddGold(500);
+        completingFaction.Player.AddLumber(2000);
+        _proudmooreCapitalShip.Rescue(completingFaction.Player);
+      }
+      else
+      {
+        _proudmooreCapitalShip.Rescue(Player(PLAYER_NEUTRAL_AGGRESSIVE));
+      }
+      PauseUnitBJ(false, _proudmooreCapitalShip);
```

Comment:
We should avoid using BJ functions where possible since they're sometimes very inefficient. Try:
```suggestion
      _proudmooreCapitalShip.Pause(false);
```
If that extension method doesn't exist, try:
```suggestion
      PauseUnit(_proudmooreCapitalShip, false);
```

Reply:
I see.
It's hard to decide which of all the different methods to use when they all sound like they're doing bascially the same thing. 
But I see that you're opting for the exentsion methods wherever possible so I'll try to keep that in mind going forward